### PR TITLE
PS-1 : Week 10 : Completed indent tracking and item-wise purchase status whole functionality.

### PR DESCRIPTION
**Pull request by kanha638**
## Proposed changes

Created endpoint for viewing the details of drafted indent and added functionality to forward indent to some other person.

Created page that will eventually show all the details related to indent

## Types of changes

_Put an `x` in the boxes that apply_

-   [ ] Bugfix (non-breaking change which fixes an issue)
-   [X] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] Other (please describe):

## Checklist

_Put an `x` in the boxes that apply_

-   [x] My code follows the [style guidelines of this project](https://docs.google.com/document/d/1GI2Hile8UbGZ82gFe1y4wAVPcNPXip1cmXTzog1S41U/edit)
-   [x] I have performed a self-review of my own code
-   [x] I have created new branch for this pull request
-   [x] I have commented my code, particularly in hard-to-understand areas
-   [x] I have made corresponding changes to the documentation
-   [x] My changes generate no new warnings
-   [x] My changes does not break the current system and it passes all the current test cases.

## Screenshots
Screenshot 2023-04-05 at 12 22 34 AM

Screenshot 2023-04-05 at 12 22 51 AM
Screenshot 2023-04-05 at 12 23 09 AM

Please attach the screenshots of the changes made in case of change in user interface

## Other information

Any other information that is important to this pull request

### CODE REVIEW

**Review by kanha638 — COMMENTED** [review]
Reviewed code

**Comment by kanha638** [comment]
Some more changes added now these functionality also included

1. delete drafted indents for indent with multiple items
2. view filled indent list
3. view filled ident details 
4. track filled indent status


Screenshots : 
Screenshot 2023-04-05 at 12 30 56 AM
Screenshot 2023-04-05 at 12 31 42 AM
Screenshot 2023-04-05 at 12 31 55 AM
Screenshot 2023-04-05 at 12 32 10 AM

**Review by harshp0702 — COMMENTED** [review]
Approved. @akshatnema

File: FusionIIIT/applications/ps1/views.py [hunk 1]
```diff
@@ -802,6 +863,139 @@ def forwardindent(request, id):
...
+#     designations = HoldsDesignation.objects.select_related('user','working','designation').filter(user=request.user)
+
+#     context = {
+#         'items':items,
+#         'designations':designations,
+#         'file': file,
+#         'track': track,
+#         'indent':indent,
+#     }
+# 	return render(request, 'ps1/createdindent.html', context)
```
Comment (akshatnema) [inline FusionIIIT/applications/ps1/views.py]: Kindly remove this comment.

File: FusionIIIT/templates/ps1/draftedIndentDetails.html [hunk 2]
```diff
@@ -0,0 +1,441 @@
...
+                                                            </a>
+                                                        </a>
+                                                        <br>
+                                                        <div class="ui divider"></div>
+                                            <center style="color: dimgrey">FileFlow is possible only when file is forwarded to other employee.</center
+                                                </div>
+                                            </div>
+                                        </div>
+                               </div>
+                        </div>-->
```
Comment (akshatnema) [inline FusionIIIT/templates/ps1/draftedIndentDetails.html]: remove the comment.

File: FusionIIIT/templates/ps1/draftedIndentDetails.html [hunk 3]
```diff
@@ -0,0 +1,441 @@
...
+
+
+<!-- <script type="text/javascript">
+    function ShowHideDiv() {
+        var chkAddress = document.getElementById("chkAddress");
+        var leaveAddress = document.getElementById("leaveAddress");
+        leaveAddress.style.display = chkAddress.checked ? "block" : "none";
+    }
+</script>
+ -->
```
Comment (akshatnema) [inline FusionIIIT/templates/ps1/draftedIndentDetails.html]: Remove this comment.

**Review by akshatnema — CHANGES_REQUESTED** [review]
Changes requested.

**Review by kanha638 — COMMENTED** [review]
@akshatnema changes done

**Comment by kanha638** [comment]
@akshatnema Please review the changes and merge it as we have to show its status on 214 server.

File: FusionIIIT/templates/ps1/viewMyIndentFile.html [hunk 4]
```diff
@@ -0,0 +1,336 @@
...
+
+
+<!-- <script type="text/javascript">
+    function ShowHideDiv() {
+        var chkAddress = document.getElementById("chkAddress");
+        var leaveAddress = document.getElementById("leaveAddress");
+        leaveAddress.style.display = chkAddress.checked ? "block" : "none";
+    }
+</script>
+ -->
```
Comment (akshatnema) [inline FusionIIIT/templates/ps1/viewMyIndentFile.html]:
```suggestion
```

File: FusionIIIT/applications/ps1/views.py [hunk 5]
```diff
@@ -802,6 +863,113 @@ def forwardindent(request, id):
...
+                receiver = request.POST.get('receiver')
+                try:
+                    receiver_id = User.objects.get(username=receiver)
+                except Exception as e:
+                    messages.error(request, 'Enter a valid destination')
+                    designations = HoldsDesignation.objects.select_related('user','working','designation').filter(user=request.user)
+
+                    context = {
+                        # 'extrainfo': extrainfo,
+                        # 'holdsdesignations': holdsdesignations,
```
Comment (akshatnema) [inline FusionIIIT/applications/ps1/views.py]:
```suggestion
```

File: FusionIIIT/applications/ps1/views.py [hunk 6]
```diff
@@ -802,6 +863,113 @@ def forwardindent(request, id):
...
+                receive = request.POST.get('recieve')
+                try:
+                    receive_design = Designation.objects.get(name=receive)
+                except Exception as e:
+                    messages.error(request, 'Enter a valid Designation')
+                    designations = HoldsDesignation.objects.select_related('user','working','designation').filter(user=request.user)
+
+                    context = {
+                        # 'extrainfo': extrainfo,
+                        # 'holdsdesignations': holdsdesignations,
```
Comment (akshatnema) [inline FusionIIIT/applications/ps1/views.py]:
```suggestion
```

File: FusionIIIT/applications/ps1/views.py [hunk 7]
```diff
@@ -802,6 +863,113 @@ def forwardindent(request, id):
...
+                        'designations': designations,
+                        'file': file,
+                        'track': track,
+                    }
+                    return render(request, 'ps1/createdindent.html', context)
+
+                # receive_design = receive_designation[0]
```
Comment (akshatnema) [inline FusionIIIT/applications/ps1/views.py:1335]:
```suggestion
```

File: FusionIIIT/applications/ps1/views.py [hunk 8]
```diff
@@ -802,6 +863,113 @@ def forwardindent(request, id):
...
+                upload_file = request.FILES.get('myfile')
+                # return HttpResponse ("success")
```
Comment (akshatnema) [inline FusionIIIT/applications/ps1/views.py:1337]:
```suggestion
```

File: FusionIIIT/templates/ps1/draftedIndentDetails.html [hunk 9]
```diff
@@ -0,0 +1,410 @@
...
+                                <!--<div class="field">
+                                    <label>Forward To:
+                                        <select required class="ui fluid search selection dropdown" name="receiver">
+                                            <option selected hidden>Select</option>
+                                            {% for holdsdesignations in holdsdesignations %}
+                                            <option value="{{holdsdesignations.user.id}}">{{holdsdesignations}}</option>
+                                            {% endfor %}
+                                        </select>
+                                    </label>
+                                </div>-->
```
Comment (akshatnema) [inline FusionIIIT/templates/ps1/draftedIndentDetails.html]:
```suggestion
```

File: FusionIIIT/templates/ps1/draftedIndentDetails.html [hunk 10]
```diff
@@ -0,0 +1,410 @@
...
+<!--                                                     <a class="author">{{file.uploader.user}}-->
+<!--                                                         <a class="metadata">-->
+<!--                                                            <span class="date">{{file.upload_date}}</span>-->
+<!--                                                         </a>-->
+<!--                                                     </a>-->
+<!--                                    </div>-->
+<!--                                                </div>-->
+
+<!--                                        </div>-->
+<!--                                    </div>-->
```
Comment (akshatnema) [inline FusionIIIT/templates/ps1/draftedIndentDetails.html]:
```suggestion
```

**Review by akshatnema — CHANGES_REQUESTED** [review]

**Comment by kanha638** [comment]
@PranshuNayak , @samay-rgb Please Review changes on DB.

**Review by harshp0702 — COMMENTED** [review]
Approved. @akshatnema

**Review by samay-rgb — APPROVED** [review]

**Review by PranshuNayak — APPROVED** [review]
@akshatnema changes in models.py are fine.

**Pull request merged**